### PR TITLE
Status bar component added

### PR DESCRIPTION
**Pull request by grumdrig**
I added a status bar component - here's a motivating screenshot of a win 98 file view:
http://toastytech.com/guis/win98desk.png

The status bar can go anywhere but it makes the most sense added to a window (in your sense of a window), so maybe it should be presented as such in the documentation. I guess I think that would be best so I can change it if you like.

Another judgement call, I've used table and td elements to implement it. You could do it with div's with display style to table and table-cell, but it's a little more tortured to do it that way so I think the way I've done it is best.

### CODE REVIEW

**Comment by grumdrig** [comment]
(Btw I'm not experienced with github and pull requests and I don't understand what the failed check is all about.)

**Comment by jdan** [comment]
Hey :) Thanks for this, I'll take a closer look tomorrow.

Re: build failures, docs/index.html is a generated file (from index.html.ejs), so that build failure is from a test that makes sure the build of `index.html.ejs` doesn't result in any changes. In your case it will, since running `npm run build` undoes your changes to `index.html`. Does that make sense?

Got a confusing setup here, sorry :) Will tidy it up.

**Comment by grumdrig** [comment]
Oh, ok. It does makes sense. I thought I'd better put an example usage
somewhere, but I guess I picked the wrong spot. Thanks for the explanation.

**Comment by jdan** [comment]
Of course :)

File: docs/index.html [hunk 1]
```diff
@@ -747,6 +748,41 @@ <h4 id="window-contents">Window contents</h4>
...
+         <div class="example" style="width:400px">
+            <table class="status-bar">
+              <td>12 object(s)
+              <td>5.47MB
+              <td>&#x1f4bb; My Computer
+            </table>
+          </div>
+          <details>
+            <summary>Show code</summary>
+            <pre><code>&lt;table class="status-bar">
```
Comment (okoflo) [inline docs/index.html:777]:
```suggestion
            <pre><code>&lt;table class="status-bar"&gt;
```
Nitpick: Style consistency.

Comment (okoflo) [reply docs/index.html:777]: Just read through the other comments however, might not need to apply this 😄 Depending on how `index.html` is being handled.

**Comment by piranna** [comment]
What about using `footer` instead of a `div`? It will be anyway at the bottom of the window...

> The status bar can go anywhere but it makes the most sense added to a window (in your sense of a window), so maybe it should be presented as such in the documentation. I guess I think that would be best so I can change it if you like.

Maybe you could do the CSS selectors to force they are under a window, like `.window > footer`...

**Comment by grumdrig** [comment]
> What about using `footer` instead of a `div`? It will be anyway at the bottom of the window...

Not necessarily a bad idea, but there's no need to require it to be in the bottom of a window, and you'd still face the issue of having to force table layout to get the panels to fill in the space correctly (or in the way I think they should). (Or solve that some other way.)

**Comment by grumdrig** [comment]
I think the documentation changes are where they belong now. 

I didn't move the example to within the window section, but I will if you'd like me to.

**Comment by piranna** [comment]
> ou'd still face the issue of having to force table layout to get the panels to fill in the space correctly

What about using `flex` layout?

**Comment by jdan** [comment]
@grumdrig I'm inclined to agree with @piranna - wanna give this a whirl as a flex-box layout? I did a quick mock-up here: https://codepen.io/jdan/pen/bGVRQBq

I'm happy to add docs for this once it's ready. Thanks for your continued effort here, much appreciated.

**Comment by grumdrig** [comment]
I think the footer with flex layout worked out pretty well. I tried to find a balance between aesthetics and authenticity with `margin` on the `footer`, and `padding` on the panels (`footer > *`), and they go a pixel or two either way. I left the example in the docs as a test case for you to replace or flesh out as you like.

**Comment by jdan** [comment]
Hey wanna rebase again? Just so the preview deployment works.

Code looks good, I'll write up some docs.

**Comment by jdan** [comment]
Looks like this closed because I deleted the master branch in favor of `main`, sorry! Feel free to re-open continuing our discussion

**Comment by grumdrig** [comment]
Okay, if I did everything right, it should be ready to go

**Pull request closed without merging**